### PR TITLE
refactor cldaero_uptakerate

### PR DESCRIPTION
**Pull request by tangshuaiqi**
relatively simple function. mainly change the organization and revising comments.

### CODE REVIEW

File: components/eam/src/chemistry/aerosol/cldaero_mod.F90 [hunk 1]
```diff
@@ -79,49 +79,60 @@ end subroutine cldaero_deallocate
...
-! apply bounds to rad_cd to avoid the occasional unphysical value
-        if (radxnum_cd .le. volx34pi_cd*4.0e4_r8) then
+! rad_cd = (drop radius in cm), computed from liquid water and drop number,
+! then bounded by 0.5 and 50.0 micrometers to avoid the occasional unphysical value
+        if (radxnum_cd <= volx34pi_cd*4.0e4_r8) then
             radxnum_cd = volx34pi_cd*4.0e4_r8
             rad_cd = 50.0e-4_r8
-        else if (radxnum_cd .ge. volx34pi_cd*4.0e8_r8) then
+        elseif (radxnum_cd >= volx34pi_cd*4.0e8_r8) then
             radxnum_cd = volx34pi_cd*4.0e8_r8
```
Comment (mingxuanwupnnl) [inline components/eam/src/chemistry/aerosol/cldaero_mod.F90]: Maybe also make 4.0e4_r8 as parameter.

File: components/eam/src/chemistry/aerosol/cldaero_mod.F90 [hunk 2]
```diff
@@ -144,5 +155,6 @@ function cldaero_uptakerate( xl, cldnum, cfact, cldfrc, tfld,  press ) result( u
         uptkrate = 12.56637_r8*radxnum_cd*gasdiffus*fuchs_sutugin
```
Comment (mingxuanwupnnl) [inline components/eam/src/chemistry/aerosol/cldaero_mod.F90]: To me, we can make 12.56637_r8 a parameter.

File: components/eam/src/chemistry/aerosol/cldaero_mod.F90 [hunk 3]
```diff
@@ -79,49 +79,60 @@ end subroutine cldaero_deallocate
...
+    real(r8) :: gasspeed        ! H2SO4 gas mean molecular speed [cm/s]
+    real(r8) :: knudsen         ! knudsen number [unitless]
+    real(r8) :: fuchs_sutugin   ! another dimensionless number
+    real(r8) :: volx34pi_cd     ! droplet volume * 3/4*pi [cm^3/cm^3]
+
+    real(r8),parameter :: one_third = 0.3333333_r8
+
+! change drop number conc from #/kg to #/cm^3
         num_cd = 1.0e-3_r8*cldnum*cfact/cldfrc
         num_cd = max( num_cd, 0.0_r8 )
```
Comment (mingxuanwupnnl) [inline components/eam/src/chemistry/aerosol/cldaero_mod.F90]: We can make 1.0e-3_r8 a unit conversion parameter. It think it is widely used and might be eventually made as a common parameter for all subroutines.

**Review by mingxuanwupnnl — APPROVED** [review]
Looks good to me.

**Review by singhbalwinder — APPROVED** [review]
Looks good! There are a lot of free numbers in this function. You can select and make some of them parameters if you like (especially the ones that repeat).

File: components/eam/src/chemistry/aerosol/cldaero_mod.F90 (hunk 2 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/aerosol/cldaero_mod.F90]: some of these numbers only used once and I don't quite understand their physical meaning and if they have any units. I will make those variables that I know that they are as parameters and keep others as it. If you know what this number mean and think it better to be made as parameter, please let me know.

Comment (brigaudet) [reply components/eam/src/chemistry/aerosol/cldaero_mod.F90]: I think this is supposed to be 4 * pi.

Comment (tangshuaiqi) [reply components/eam/src/chemistry/aerosol/cldaero_mod.F90]: aha, thank you @brigaudet . I will add it as parameter

**Comment by tangshuaiqi** [comment]
Since there is no further comments and all tests are passed. I will merge this PR.

  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel

**Pull request merged**